msprime/simulations.py

```diff
@@ -1517,6 +1517,9 @@ class SimulationModelChange(DemographicEvent):
     :param float time: The time at which the simulation model changes
         to the new model, in generations. After this time, all internal
         tree nodes, edges and migrations are the result of the new model.
+        If time is set to infinity using ``np.Inf`` this has the effect
```

Comment:
It's not particularly intuitive though. It would be better if we made ``time=None`` the default, and then said "If time is unspecified or None, the model change occurs immediately after the previous model completes". We need to make this work for chaining multiple sweeps together too and I don't think the np.inf thing is going to work there, we'll have to detect None explicitly in the model change logic.